mayastor/tests/error_count.rs

```diff
@@ -0,0 +1,232 @@
+    Reactor::block_on(async {
+        inject_error(SPDK_BDEV_IO_TYPE_READ, VBDEV_IO_FAILURE, 257).await;
+        inject_error(SPDK_BDEV_IO_TYPE_WRITE, VBDEV_IO_FAILURE, 100).await;
+        for _ in 0 .. 257 {
+            err_read_nexus().await; // multiple reads because there are two replicas
+            err_read_nexus().await; // and we may get the wrong one
+        }
+        for _ in 0 .. 100 {
+            err_write_nexus().await;
+        }
+    });
+
+    reactor_pause_millis(1); // give time for any errors to be added to the error store
+
+    nexus_err_query_and_test(
+        BDEV_EE_ERROR_DEVICE,
+        NexusErrStore::READ_FLAG,
+        156,
+    );
+    nexus_err_query_and_test(
+        BDEV_EE_ERROR_DEVICE,
+        NexusErrStore::WRITE_FLAG,
+        100,
+    );
+
+    mayastor_env_stop(0);
+}
+
+async fn inject_error(op: u32, mode: u32, count: u32) {
+    let retval: i32;
+    let err_bdev_name_str =
+        CString::new(EE_ERROR_DEVICE).expect("Failed to create name string");
+    let raw = err_bdev_name_str.into_raw();
+
+    unsafe {
+        retval = spdk_vbdev_error_inject_error(raw, op, mode, count);
+    }
+    assert_eq!(retval, 0);
+}
+
+async fn create_error_bdev() {
+    let mut retval: i32;
+    let cname = CString::new(ERROR_DEVICE).unwrap();
+    let filename = CString::new(DISKNAME2).unwrap();
+
+    unsafe {
+        // this allows us to create a bdev without its name being a uri
+        retval = create_aio_bdev(cname.as_ptr(), filename.as_ptr(), 512)
+    };
+    assert_eq!(retval, 0);
+
+    let err_bdev_name_str = CString::new(ERROR_DEVICE.to_string())
+        .expect("Failed to create name string");
+    unsafe {
+        retval = spdk_vbdev_error_create(err_bdev_name_str.as_ptr()); // create the error bdev around it
+    }
+    assert_eq!(retval, 0);
+}
+
+async fn create_nexus() {
+    let ch = vec![BDEVNAME1.to_string(), BDEV_EE_ERROR_DEVICE.to_string()];
+
+    nexus_create(ERROR_COUNT_TEST_NEXUS, 64 * 1024 * 1024, None, &ch)
+        .await
+        .unwrap();
+}
+
+fn nexus_err_query_and_test(
+    child_bdev: &str,
+    io_type_flags: u32,
+    expected_count: u32,
+) {
+    let nexus = nexus_lookup(ERROR_COUNT_TEST_NEXUS).unwrap();
+    let count = nexus
+        .error_record_query(
+            child_bdev,
+            io_type_flags,
+            NexusErrStore::IO_FAILED_FLAG,
+            1_000_000_000, // within the past 1 second
+        )
+        .expect("failed to query child");
+    assert!(count.is_some()); // true if the error_store is enabled
+    assert_eq!(count.unwrap(), expected_count);
+}
+
+async fn err_write_nexus() {
+    let bdev = Bdev::lookup_by_name(ERROR_COUNT_TEST_NEXUS)
+        .expect("failed to lookup nexus");
+    let d = bdev
+        .open(true)
+        .expect("failed open bdev")
+        .into_handle()
+        .unwrap();
+    let buf = d.dma_malloc(512).expect("failed to allocate buffer");
+
+    let _ = d.write_at(0, &buf).await;
```

Comment:
should this assert that it actually got an error?

Reply:
see below